Hi @TomHodson, thank you for adding this feature and the tests. It all looks very good. However, when I use your fork the tests fail for me:

These are my setup steps:
```
git clone git@github.com:TomHodson/findlibs.git findlibs_pr
cd findlibs_pr
git checkout remotes/origin/TomHodson-add-extra-global-lib-path
pip install -e ".[test]"
pytest
```

This is my output:
```
(mpy10) √ findlibs_pr % pytest
============================================================== test session starts ===============================================================
platform darwin -- Python 3.10.12, pytest-7.3.2, pluggy-1.2.0
rootdir: /Users/cgr/git/findlibs_pr
plugins: pyfakefs-5.2.2, anyio-3.7.0
collected 7 items

tests/test_config_file.py .FF....                                                                                                          [100%]

==================================================================== FAILURES ====================================================================
_____________________________________________________________ test_empty_config_file _____________________________________________________________

fs = <pyfakefs.fake_filesystem.FakeFilesystem object at 0x117231b40>

    def test_empty_config_file(fs):
        "Check that it works with an empty config file"
        fs.create_file(testlib_path)
        fs.create_file(config_path)
>       assert config_path.exists()
E       AssertionError: assert False
E        +  where False = <bound method Path.exists of PosixPath('/Users/cgr/.findlibs.yml')>()
E        +    where <bound method Path.exists of PosixPath('/Users/cgr/.findlibs.yml')> = PosixPath('/Users/cgr/.findlibs.yml').exists

/Users/cgr/git/findlibs_pr/tests/test_config_file.py:20: AssertionError
____________________________________________________________ test_empty_search_paths _____________________________________________________________

fs = <pyfakefs.fake_filesystem.FakeFilesystem object at 0x11729e920>

    def test_empty_search_paths(fs):
        "Check that it works with an empty config file"
        fs.create_file(testlib_path)
        fs.create_file(config_path, contents =
    """
    additional_search_paths:
    """)

>       assert config_path.exists()
E       AssertionError: assert False
E        +  where False = <bound method Path.exists of PosixPath('/Users/cgr/.findlibs.yml')>()
E        +    where <bound method Path.exists of PosixPath('/Users/cgr/.findlibs.yml')> = PosixPath('/Users/cgr/.findlibs.yml').exists

/Users/cgr/git/findlibs_pr/tests/test_config_file.py:31: AssertionError
============================================================ short test summary info =============================================================
FAILED tests/test_config_file.py::test_empty_config_file - AssertionError: assert False
FAILED tests/test_config_file.py::test_empty_search_paths - AssertionError: assert False
```

--- a/findlibs/__init__.py
+++ b/findlibs/__init__.py
@@ -8,10 +8,26 @@
 # nor does it submit to any jurisdiction.
 #
 
+"""
+Use python config parser
+
+file locations: 
+~/.findlibs, ~/.config/findlibs/findlibs.conf
+if both, throw error
+
+config file: 
+[Paths]
+/path/to/foo
+/path/to/bar
+/path/to/baz
+
+if relative paths or files, throw error
+"""
+
 import ctypes.util
 import os
 import sys
-import yaml
+import configparser
 from pathlib import Path
 
 __version__ = "0.0.5"
@@ -20,6 +36,44 @@ EXTENSIONS = {
     "darwin": ".dylib",
     "win32": ".dll",
 }
+
+def _get_paths_from_config():
+    locations = [Path(p).expanduser() for p in [
+        "~/.config/findlibs/findlibs.conf",
+        "~/.findlibs",
+    ]]
+
+    locations = [p for p in locations if p.exists()]
+
+    if len(locations) == 0: return []
+    if len(locations) > 1:     
+        raise ValueError(f"There are multiple config files! Delete all but one of {locations}")
+
+    config = configparser.RawConfigParser(allow_no_value = True) # Allow keys without values
+    config.optionxform = lambda option: option # Preserve case of keys 
+    
+    with open(locations[0], "r") as f:
+        config.read(locations[0])
+    
+    if "Paths" not in config: return []
+    print(config)
+    # replace $HOME with ~, expand ~ to full path, 
+    # resolve any relative paths to absolute paths
+    paths = {Path(p.replace("$HOME", "~")).expanduser()
+            for p in config["Paths"] or []}
+    
+    relative_paths = [p for p in paths if not p.is_absolute()]
+    if relative_paths:
+        raise ValueError(f"Don't use relative paths in the config file ({locations[0]}), offending paths are: {relative_paths}")
+    
+    files = [p for p in paths if not p.is_dir()]
+    if files:
+        raise ValueError(f"Don't put files in the config file ({locations[0]}), offending files are: {files}")
+    
+
+    return paths
+
+
 
 
 def find(lib_name, pkg_name=None):
@@ -36,6 +90,7 @@ def find(lib_name, pkg_name=None):
 
     pkg_name = pkg_name or lib_name
     extension = EXTENSIONS.get(sys.platform, ".so")
+    libname = "lib{}{}".format(lib_name, extension)
 
     # sys.prefix/lib, $CONDA_PREFIX/lib has highest priority;
     # otherwise, system library may mess up anaconda's virtual environment.
@@ -46,7 +101,8 @@ def find(lib_name, pkg_name=None):
 
     for root in roots:
         for lib in ("lib", "lib64"):
-            fullname = os.path.join(root, lib, "lib{}{}".format(lib_name, extension))
+            fullname = os.path.join(root, lib, libname)
+            print(f"looking at {fullname}")
             if os.path.exists(fullname):
                 return fullname
 
@@ -59,41 +115,30 @@ def find(lib_name, pkg_name=None):
             home = os.path.expanduser(os.environ[env])
             for lib in ("lib", "lib64"):
                 fullname = os.path.join(
-                    home, lib, "lib{}{}".format(lib_name, extension)
+                    home, lib, libname
                 )
                 if os.path.exists(fullname):
                     return fullname
 
-    config_file = Path("~/.findlibs.yml").expanduser()
+    config_paths = _get_paths_from_config()
 
-    if config_file.exists():
-        with open(config_file, "r") as f:
-            config = yaml.safe_load(f.read()) or dict()
-            
-        search_paths = config.get("additional_search_paths", []) or []
-            
-        # replace $HOME with ~, expand ~ to full path, 
-        # resolve any relative paths to absolute paths
-        config_paths = {Path(p.replace("$HOME", "~")).expanduser().resolve() 
-                for p in search_paths}
-
-        for root in config_paths:
-            for lib in ("lib", "lib64"):        
-                filepath = root / lib / f"lib{lib_name}{extension}"
-                if filepath.exists(): return str(filepath)
+    for root in config_paths:
+        for lib in ("lib", "lib64"):        
+            filepath = root / lib / f"lib{lib_name}{extension}"
+            if filepath.exists(): return str(filepath)
 
     for path in (
         "LD_LIBRARY_PATH",
         "DYLD_LIBRARY_PATH",
     ):
         for home in os.environ.get(path, "").split(":"):
-            fullname = os.path.join(home, "lib{}{}".format(lib_name, extension))
+            fullname = os.path.join(home, libname)
             if os.path.exists(fullname):
                 return fullname
 
     for root in ("/", "/usr/", "/usr/local/", "/opt/", "/opt/homebrew/", os.path.expanduser("~/.local")):
         for lib in ("lib", "lib64"):
-            fullname = os.path.join(root, lib, "lib{}{}".format(lib_name, extension))
+            fullname = os.path.join(root, lib, libname)
             if os.path.exists(fullname):
                 return fullname
 

--- a/findlibs/__init__.py
+++ b/findlibs/__init__.py
@@ -8,22 +8,6 @@
 # nor does it submit to any jurisdiction.
 #
 
-"""
-Use python config parser
-
-file locations: 
-~/.findlibs, ~/.config/findlibs/findlibs.conf
-if both, throw error
-
-config file: 
-[Paths]
-/path/to/foo
-/path/to/bar
-/path/to/baz
-
-if relative paths or files, throw error
-"""
-
 import ctypes.util
 import os
 import sys

--- a/setup.py
+++ b/setup.py
@@ -41,7 +41,7 @@ setuptools.setup(
     url="https://github.com/ecmwf/findlibs",
     packages=setuptools.find_packages(),
     include_package_data=True,
-    install_requires=["PyYAML"],
+    install_requires=[],
     extras_require={
         'test': ["pytest", "pyfakefs"]
     },

--- a/tests/test_config_file.py
+++ b/tests/test_config_file.py
@@ -6,12 +6,43 @@ from pathlib import Path
 
 extension = findlibs.EXTENSIONS.get(sys.platform, ".so")
 testlib_path = Path(f"/usr/lib/test{extension}")
-config_path = Path(f"~/.findlibs.yml").expanduser().resolve()
+config_path = Path(f"~/.config/findlibs/findlibs.conf").expanduser().resolve()
+
+config_paths = [Path(f"~/.config/findlibs/findlibs.conf").expanduser().resolve(),
+                Path(f"~/.findlibs").expanduser().resolve()]
 
 def test_no_config_file(fs):
     "Check that findlibs works with no config file"
     fs.create_file(testlib_path)
     assert findlibs.find("test") == str(testlib_path)
+
+def test_both_config_files(fs):
+    "Check that it throws an error if two config files are present"
+    for p in config_paths: fs.create_file(p)
+    with pytest.raises(ValueError):
+        findlibs._get_paths_from_config()
+
+def test_relative_path(fs):
+    "Check that it throws an error on relative paths"
+    fs.create_file(config_path, contents = 
+"""
+[Paths]
+relative/path
+""")
+
+    with pytest.raises(ValueError):
+        findlibs._get_paths_from_config()
+
+def test_file(fs):
+    "Check that it throws an error when files are included"
+    fs.create_file(config_path, contents = 
+"""
+[Paths]
+/path/to/file.so
+""")
+
+    with pytest.raises(ValueError):
+        findlibs._get_paths_from_config()
 
 def test_empty_config_file(fs):
     "Check that it works with an empty config file"
@@ -25,7 +56,7 @@ def test_empty_search_paths(fs):
     fs.create_file(testlib_path)
     fs.create_file(config_path, contents = 
 """
-additional_search_paths:
+[Paths]
 """)
 
     assert config_path.exists()
@@ -45,8 +76,8 @@ def test_config(fs, search_dir, testlib_path):
     fs.create_file(testlib_path)
     fs.create_file(config_path, contents = 
 f"""
-additional_search_paths:
-  - {search_dir}
+[Paths]
+{search_dir}
 """)
     assert findlibs.find("test") == str(testlib_path)
     

